lib/view_component/deprecation.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module ViewComponent
+  # ViewComponent should warn consumers when they are using a deprecated method.
+  # In local development, however, those deprecation warnings are distracting.
+  # This class wraps the deprecation warning functionality in some basic logic
+  # so the deprecation warnings are only shown to consumers.
+  class Deprecation
+    def self.warn(message)
+      new.warn(message)
+    end
+
+    def initialize
+      @deprecation_class = if local_test_env?
+                             NoopDeprecation
+                           else
+                             ActiveSupport::Deprecation
+                           end
```

Comment:
```suggestion
      @deprecation_class = if local_test_env?
        NoopDeprecation
      else
        ActiveSupport::Deprecation
      end
```

Thoughts on keeping the indentation consistent for readability?

Reply:
TBH I could go either way. The original, typed indentation was as you suggest but vim's `V}=` moved it to this so ¯\_(ツ)_/¯. Happy to defer to @joelhawksley on that one